src/main/java/com/bobocode/petros/container/ApplicationAnnotationContainer.java

```diff
@@ -35,20 +33,26 @@ public ApplicationAnnotationContainer(String packageName) {
     @Override
     public <T> T getDependency(String name, Class<T> clazz) {
         LOG.debug("Searching dependency by name {} and by type {}", name, clazz);
-        DependencyDefinition definition = keyByDependencyDefinitionName(name);
+        DependencyDefinition definition = new DependencyDefinition();
+        try {
+            definition = keyByDependencyDefinitionName(name);
+        } catch (Exception e) {
+            LOG.error(e.getMessage(), e);
+        }
         LOG.debug("Dependency definition found {}", definition);
         return (T) dependencyMap.get(definition);
     }
 
 
     @Override
     public <T> T getDependency(Class<T> clazz) {
-        if (isNonUniqueDependency(clazz)) {
-            LOG.debug("The dependency with such {} type already exists in container", clazz.getName());
-            throw new NoUniqueDependecyException(clazz.getName());
-        } else {
-            return getDependencyFromMap(clazz);
+        T dependency = null;
+        try {
+            dependency = getDependencyFromMap(clazz);
+        } catch (NoUniqueDependencyException e) {
```

Comment:
method `getDependencyFromMap` can also throw NosuchElementException . I guess we should pass it to this level and handle it with a meaningful message, not just a default one -  "no value present'. What do you think?

Reply:
Geeeeez, I've missed that, thanks)